src/ui/actions/timeline.ts

```diff
@@ -500,3 +516,40 @@ export function clearHoveredItem(): UIThunkAction {
 export function setPlaybackPrecachedTime(time: number): SetPlaybackPrecachedTimeAction {
   return { type: "set_playback_precached_time", time };
 }
+
+export function setTrimRegion(trimRegion: {
+  startTime: number;
+  endTime: number;
+}): SetTrimRegionAction {
+  return { type: "set_trim_region", trimRegion };
+}
+
+export function updateTrimRegion(
+  type: "startTime" | "endTime" | "span",
+  midpointDifference: number
+): UIThunkAction {
+  return ({ dispatch, getState }) => {
+    const state = getState();
+    const hoverTime = getHoverTime(state);
+    const trimRegion = getTrimRegion(state);
+    const zoomRegion = getZoomRegion(state);
+
+    if (!hoverTime || !trimRegion) return;
+
+    if (type === "span") {
+      const { startTime, endTime } = trimRegion;
+      const oldSpanMidpoint = (endTime + startTime) / 2;
```

Comment:
I would maybe pull this and the clamp function out into well-named utils like `timelineMidPoint(start, end)` and `clampToZoomRegion(zoomRegion, time)` but YMMV!

Reply:
Looks like a good place could be `utils/timeline`